plugins/main/bridge/bridge.go

```diff
@@ -236,6 +238,17 @@ func ensureBridge(brName string, mtu int, promiscMode, vlanFiltering bool) (*net
 		return nil, fmt.Errorf("could not add %q: %v", brName, err)
 	}
 
+	// check if master was specified for the bridge
+	if master != "" {
+		m, err := netlink.LinkByName(master)
+		if err != nil {
+			return nil, fmt.Errorf("failed to lookup master %q: %v", master, err)
```

Comment:
There is missing teardown.
If during this particular run of plugin bridge interface was created (err != nil && err != syscall.EEXIST) - the bridge interface should be removed.
It's not a big issue because... later in this func the same teardown procedure is also missing in case of any error...

Reply:
Thanks @jellonek for that recommendation! I added the teardown for the block and also for the others lower into the function and verified that the bridge does indeed get deleted when an error is thrown